apis/v1alpha1/iam_request.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+// IamRequest represents a request to update IAM policies.
+type IamRequest struct {
+	// List of ResourcePolicy, each specifies the IAM principals/members to role
+	// bindings to be added for a GCP resource IAM policy.
+	ResourcePolicies []*ResourcePolicy `json:"policies,omitempty"`
+}
+
+// ResourcePolicy specifies the IAM principals/members to role bindings to be
+// added for a GCP resource IAM policy.
+type ResourcePolicy struct {
```

Comment:
I think we can also have "conditions" here?

Reply:
This is for parsing the IAM request which only contains roles and members. Condition is for permission expiry, where we would obtain the duration from PR labels according to the design, and is added when we call GCP API to update the policy.

Reply:
How come? We don't want a requester to be able to set expiry themselves.

Reply:
Conditions aren't just for expiration; they can be used to define context-aware access and other attributes.

Reply:
I'd say leave the condition out for now. If we introduce it now, we need to look into `Expr` https://pkg.go.dev/google.golang.org/api/cloudresourcemanager/v3#Expr and find whether it's a expiry condition and disallow that.
We can either try to support the condition later or leave it to the `gcloud` command request

Reply:
Per offline discussion, we agreed we can leave the condition out at the moment.